### PR TITLE
WEA-17 Minor code refactor and feature addition.

### PR DESCRIPTION
**Pull request by ayush-shta**
1. Circular background for radio selected.
2. Provide latitude longitude in search location instead of address
3. Circular gradient background instead of linear
4. Option to get current location from drawer.

@subash-poudel

### CODE REVIEW

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 1]
```diff
@@ -102,6 +104,11 @@
     }
   }
 
+  @OnClick(R.id.tv_current_location) void fetchByCurrentLocation() {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: onCurrentLocationButtonClicked

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 2]
```diff
@@ -102,6 +104,11 @@
...
+    presenter.fetchCurrentLocation();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: presenter.onCurrentLocationButtonClicked()

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 3]
```diff
@@ -269,7 +276,10 @@ public void onPermissionsDenied(int requestCode, @NonNull List<String> perms) {
     if (requestCode == PLACE_AUTOCOMPLETE_REQUEST_CODE) {
       if (resultCode == RESULT_OK) {
         Place place = PlaceAutocomplete.getPlace(this, data);
-        presenter.searchLocation((String) place.getAddress());
+        LatLng latLng = place.getLatLng();
+        double lat = latLng.latitude;
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: redundant variable declaration

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 4]
```diff
@@ -40,7 +37,8 @@
 public class HomePresenterImpl implements HomeContract.Presenter {
   private final static int LOCATION_REQ_INTERVAL = 10000;
   private final static int FASTEST_LOCATION_REQ_INTERVAL = 5000;
-  private final static int LOCALITY_INDEX = 0;
+  private final static String ADDRESS_ROUTE = "route";
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: private static final

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 5]
```diff
@@ -162,10 +159,19 @@ public void onResponse(@NonNull Call<GeoLocation> call,
 
         if (addressList != null && !addressList.isEmpty()) {
           List<AddressComponents> addressComponentsList = addressList.get(0).getAddressComponents();
-          localityAddress = addressComponentsList.get(LOCALITY_INDEX).getLongName();
-          view.setLocality(localityAddress);
+          for (AddressComponents addressComponent : addressComponentsList) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: fishy for loop. view could set the same locality twice

File: app/src/main/res/drawable/background_gradient_rainy.xml [hunk 6]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle" >
   <gradient
-      android:angle="90"
-      android:startColor="#C8E4F8"
-      android:centerColor="#C8E4F8"
-      android:endColor="#7CD9F5"
-      android:type="linear" />
+      android:gradientRadius="140%"
```
Comment (subash-poudel) [inline app/src/main/res/drawable/background_gradient_rainy.xml:4]: why is it 140% and not 100%? See this:
https://stackoverflow.com/questions/11176590/gradient-radius-as-percentage-of-screen-size

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 1 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java (hunk 4 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 7]
```diff
@@ -176,36 +177,53 @@ public void onFailure(@NonNull Call<GeoLocation> call, @NonNull Throwable t) {
...
-          List<AddressComponents> addressComponentsList = addressList.get(0).getAddressComponents();
-          localityAddress = addressComponentsList.get(LOCALITY_INDEX).getLongName();
-          view.setLocality(localityAddress);
-        } else {
-          Toast.makeText(getContext(), "Location invalid", Toast.LENGTH_SHORT).show();
-        }
+  private String getAddressPrimary(List<AddressComponents> addressComponentsList) {
+    for (AddressComponents addressComponent : addressComponentsList) {
+      if (addressComponent.getTypes().contains(ADDRESS_STREET)) {
+        return addressComponent.getShortName();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: ??

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 8]
```diff
@@ -176,36 +177,53 @@ public void onFailure(@NonNull Call<GeoLocation> call, @NonNull Throwable t) {
...
+    return "";
+  }
 
-      @Override public void onFailure(Call<GeoLocation> call, Throwable t) {
-        Timber.e(t);
+  private String getAddressSecondary(List<AddressComponents> addressComponentsList) {
+    for (AddressComponents addressComponent : addressComponentsList) {
+      if (addressComponent.getTypes().contains(ADDRESS_CITY)) {
+        return addressComponent.getLongName();
+      } else if (addressComponent.getTypes().contains(ADDRESS_ADMINISTRATIVE_AREA)) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: refore else

**Pull request merged**